lib/daru/vector.rb

```diff
@@ -1141,6 +1159,20 @@ def jackknife(estimators, k=1) # rubocop:disable Metrics/AbcSize,Metrics/MethodL
       Daru::DataFrame.new ps
     end
 
+    # Returns an array of either none or integer values, indicating the
+    # +regexp+ matching with the given array.
+    #
+    # @param Regexp A regular matching expression. For example, +/weeks/+.
+    # @return [Array] Conatining either +nil+ or integer values, according to the match with the given +regexp+
+    # @example
+    #   dv = Daru::Vector.new ['3 days', '5 weeks', '2 weeks']
+    #   dv.match /weeks/
+    #
+    #   # => [nil, 2, 2]
+    def match(regexp)
+      @data.map { |x| x =~ regexp }
```

Comment:
Let's use [===](http://www.rubyletter.com/blog/2017/01/18/ruby-triple-equals-operator.html) operator. This way we can also do `vector.match(1..20)` or `vector.match(Numeric)`.